connectivity/netsocket/source/TCPSocket.cpp

```diff
@@ -266,7 +267,12 @@ TCPSocket *TCPSocket::accept(nsapi_error_t *error)
         ret = _stack->socket_accept(_socket, &socket, &address);
 
         if (0 == ret) {
-            connection = new TCPSocket(this, socket, address);
+            connection = new (std::nothrow) TCPSocket(this, socket, address);
+            if (!connection) {
+                _stack->socket_close(socket);
```

Comment:
I'm not sure the passive socket must be closed in response of OOM. If memory was freed the accept tried again, it might succeed.